src/common/components/table/filters/table-filters-utils.tsx

```diff
@@ -60,6 +60,25 @@ export const useDateFilterHandler = (useShallow: boolean = true) => {
   };
 };
 
+const removeDateFilterFromSearchParams = (searchParams: URLSearchParams) => {
+  searchParams.delete('startTime');
+  searchParams.delete('endTime');
+  return searchParams;
+};
+
+export const useClearDateFilterHandler = (useShallow: boolean = true) => {
+  const searchParams = useSearchParams();
+  const shallowRouter = useShallowRouter();
+  const router = useRouter();
+  const params = new URLSearchParams(searchParams);
+  const searchParamsWithoutDateFilter = removeDateFilterFromSearchParams(params);
+  if (useShallow) {
+    return () => shallowRouter.replace(null, '', `?${searchParamsWithoutDateFilter.toString()}`);
+  } else {
+    return () => router.replace(`?${searchParamsWithoutDateFilter.toString()}`);
+  }
+};
+
 export const getAddressFilterParams = (
```

Comment:
I'm a bit unclear on the intent of this function, name suggests it's a getter but it can set or delete a params.

Reply:
I see your point. Those functions have been removed, and I have abstracted the query manipulation functionality into a hook called useQueryUpdater, and I have simplified the logic for handling query params for each filter into individual filter mutator functions.